src/garage/replay_buffer/replay_buffer.py

```diff
@@ -12,173 +12,75 @@
-            self._size_in_transitions, self._n_transitions_stored +
-            self._time_horizon * rollout_batch_size)
 
     @abstractmethod
-    def sample(self, batch_size):
-        """Sample a transition of batch_size.
+    def sample_transitions(self, batch_size):
+        """Sample N transitions, where N=batch_size.
 
         Args:
-            batch_size(int): The number of transitions to be sampled.
+            batch_size (int): The number of transitions to be sampled.
 
+        Returns:
+            SampleBatch: batch_size transitions sampled.
         """
-        raise NotImplementedError
 
-    def add_transition(self, **kwargs):
-        """Add one transition into the replay buffer.
+    def sample_path(self):
+        """Sample one path of arbitrary length.
 
-        Args:
-            kwargs (dict(str, [numpy.ndarray])): Dictionary that holds
-                the transitions.
+        Returns:
+            SampleBatch: Batch of transitions that make up a path.
 
+        Raises:
+            NotYetImplementedError: If the replay buffer does not support
+                path sampling.
         """
-        transition = {k: [v] for k, v in kwargs.items()}
-        self.add_transitions(**transition)
-
-    def add_transitions(self, **kwargs):
-        """Add multiple transitions into the replay buffer.
+        raise NotImplementedError
 
-        A transition contains one or multiple entries, e.g.
-        observation, action, reward, terminal and next_observation.
-        The same entry of all the transitions are stacked, e.g.
-        {'observation': [obs1, obs2, obs3]} where obs1 is one
-        numpy.ndarray observation from the environment.
+    @abstractmethod
+    def add_path(self, path):
+        """Add a path to the buffer.
 
         Args:
-            kwargs (dict(str, [numpy.ndarray])): Dictionary that holds
-                the transitions.
-
+            path (dict[str, np.ndarray] or TrajectoryBatch): Path to be added to the buffer.
         """
-        if not self._initialized_buffer:
-            self._initialize_buffer(**kwargs)
-
-        for key, value in kwargs.items():
-            self._episode_buffer[key].append(value)
 
-        if len(self._episode_buffer['observation']) == self._time_horizon:
-            self.store_episode()
-            for key in self._episode_buffer:
-                self._episode_buffer[key].clear()
-
-    def _initialize_buffer(self, **kwargs):
-        for key, value in kwargs.items():
-            self._episode_buffer[key] = list()
-            values = np.array(value)
-            self._buffer[key] = np.zeros(
-                [self._size, self._time_horizon, *values.shape[1:]],
-                dtype=values.dtype)
-        self._initialized_buffer = True
-
-    def _get_storage_idx(self, size_increment=1):
-        """Get the storage index for the episode to add into the buffer.
+    @abstractmethod
+    def add_transitions(self, transitions):
```

Comment:
i think this should accept only `TimeStepBatch` or possible `Union[TimeStep, TimeStepBatch]`